aspnetcore/web-api/index.md

```diff
@@ -3,14 +3,591 @@ title: Create web APIs with ASP.NET Core
+
+[!code-csharp[](index/samples/3.x/Controllers/ProductsController.cs?name=snippet_BindingSourceAttributes&highlight=3)]
+
+The `[ApiController]` attribute applies inference rules for the default data sources of action parameters. These rules save you from having to identify binding sources manually by applying attributes to the action parameters. The binding source inference rules behave as follows:
+
+* `[FromServices]` is inferred for complex type parameters registered in the DI Container.
+* `[FromBody]` is inferred for complex type parameters not registered in the DI Container. An exception to the `[FromBody]` inference rule is any complex, built-in type with a special meaning, such as <xref:Microsoft.AspNetCore.Http.IFormCollection> and <xref:System.Threading.CancellationToken>. The binding source inference code ignores those special types.
+* `[FromForm]` is inferred for action parameters of type <xref:Microsoft.AspNetCore.Http.IFormFile> and <xref:Microsoft.AspNetCore.Http.IFormFileCollection>. It's not inferred for any simple or user-defined types.
+* `[FromRoute]` is inferred for any action parameter name matching a parameter in the route template. When more than one route matches an action parameter, any route value is considered `[FromRoute]`.
+* `[FromQuery]` is inferred for any other action parameters.
+
+### FromBody inference notes
+
+`[FromBody]` isn't inferred for simple types such as `string` or `int`. Therefore, the `[FromBody]` attribute should be used for simple types when that functionality is needed.
+
+When an action has more than one parameter bound from the request body, an exception is thrown. For example, all of the following action method signatures cause an exception:
+
+* `[FromBody]` inferred on both because they're complex types.
+
+  ```csharp
+  [HttpPost]
+  public IActionResult Action1(Product product, Order order)
+  ```
+
+* `[FromBody]` attribute on one, inferred on the other because it's a complex type.
+
+  ```csharp
+  [HttpPost]
+  public IActionResult Action2(Product product, [FromBody] Order order)
+  ```
+
+* `[FromBody]` attribute on both.
+
+  ```csharp
+  [HttpPost]
+  public IActionResult Action3([FromBody] Product product, [FromBody] Order order)
+  ```
+
+<a name="FSI7"></a>
+
+### FromServices inference notes
+
+`[FromServices]` inference could, in rare cases, affect the app if a parameter type is registered in the Dependency Injection container and also expected to be bound from a different source, eg. `FromBody`.
+
+It's possible to avoid the inference by one of the following approaches:
+
+* Adding the desired binding source attribute to the Action parameter.
+* Disabling the `[FromServices]` inference rule by setting `Microsoft.AspNetCore.Mvc.ApiBehaviorOptions.DisableImplicitFromServicesParameters` to `true`:
+
+[!code-csharp[](index/samples/6.x/Program.cs?name=snippet_d400D7&highlight=14)]
```

Comment:
@brunolins16  @serpent5  please review this and compare to https://github.com/dotnet/AspNetCore.Docs/pull/25110/files
also at [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-7.0&branch=pr-en-us-25112#FSI7)